scripts/release.sh

```diff
@@ -43,31 +41,58 @@ fi
 develop_branch="develop"
 main_branch="master"
 
-# Stash any local changes to avoid errors during checkout
-git status
-prompt_user "Would you like to stash any local changes"
-if [[ $? == 1 ]]; then
-    # Stash local changes to prevent issues with checkout
-    git stash
-    echo "use \"git stash pop\" when this script is complete to restore your changes"
+# Checkout develop - so we can update versions.
+# We need to checkout the branch before we start modifying files.
+current_branch=$(git branch --show-current)
+if [ $current_branch == $develop_branch ]; then
+    echo
+    echo "$develop_branch is already checked out."
 else
-    # Dump local changes to prevent issues with checkout
-    git reset --hard
+    echo
+    echo "Checkout of $develop_branch is required for some steps"
+    prompt_user "Would you like to automatically checkout $develop_branch"
```

Comment:
1. That text is the output of the git file-diff command.  I do not know what it means.  I can filter the output, but I figured it was not critical to the work we are doing.

2. Good question.  Clearly git file-diff thinks you do have changes.  Maybe I can find a better command, or write soem code to analyze and filter the file-diff output?

Reply:
I think I have fixed this.  I use `git status` instead of `git diff-files`.  Then I use `sed` to look for a 'dirty' response from `git status`.  I wasn't sure how to manipulate the `git status` into giving me a file list, so I went ahead and added the `sed` command to the `git diff-file` that will clean up the file list.  I also briefly considered a `git clean` but I think it is unnecessary for the checkout.